### PR TITLE
refactor(start_planner): separate start planner parameters and debug data structure

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
## Description

separate start planner parameters and debug data structure

## Tests performed

test with psim

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_start_planner_module/include/behavior_path_start_planner_module/data_structs.hpp [hunk 1]
```diff
@@ -29,10 +29,30 @@
...
 using freespace_planning_algorithms::RRTStarParam;
 
+struct StartPlannerDebugData
+{
+  // filtered objects
+  PredictedObjects filtered_objects;
+  TargetObjectsOnLane target_objects_on_lane;
+  std::vector<PoseWithVelocityStamped> ego_predicted_path;
+  // collision check debug map
+  CollisionCheckDebugMap collision_check;
```
Comment (kosuke55) [inline planning/behavior_path_start_planner_module/include/behavior_path_start_planner_module/data_structs.hpp:48]: these data are just for debug?

File: planning/behavior_path_start_planner_module/include/behavior_path_start_planner_module/start_planner_module.hpp [hunk 2]
```diff
@@ -191,7 +191,7 @@ class StartPlannerModule : public SceneModuleInterface
 
   std::vector<std::shared_ptr<PullOutPlannerBase>> start_planners_;
   PullOutStatus status_;
-  mutable StartGoalPlannerData start_planner_data_;
+  mutable StartPlannerDebugData start_planner_data_;
```
Comment (kosuke55) [inline planning/behavior_path_start_planner_module/include/behavior_path_start_planner_module/start_planner_module.hpp:194]:
```suggestion
  mutable StartPlannerDebugData start_planner_debug_data_;
```

Comment (kosuke55) [inline planning/behavior_path_start_planner_module/include/behavior_path_start_planner_module/start_planner_module.hpp:194]:
```suggestion
  mutable StartPlannerDebugData debug_data_;
```

File: planning/behavior_path_start_planner_module/include/behavior_path_start_planner_module/data_structs.hpp (hunk 1 above)
Comment (kyoichi-sugahara) [reply planning/behavior_path_start_planner_module/include/behavior_path_start_planner_module/data_structs.hpp:48]: Yes!

File: planning/behavior_path_start_planner_module/include/behavior_path_start_planner_module/start_planner_module.hpp (hunk 2 above)
Comment (kyoichi-sugahara) [reply planning/behavior_path_start_planner_module/include/behavior_path_start_planner_module/start_planner_module.hpp]: Thanks!
changed in https://github.com/autowarefoundation/autoware.universe/pull/6101/commits/e6ab77a1bc3464ba3accd53cdc90e0e572394fe4

**Review by kosuke55 — APPROVED** [review]

**Pull request merged**